### ISSUE TITLE
Left Navigation Panel - white text on white background

### ISSUE THREAD

**Issue by fasson** [moneymanagerex/moneymanagerex#3364]
MMEX version:
 - [ ] 1.3.7
 - [ ] 1.3.6
 - [X] 1.5.1
Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).
-- 

Operating System:
 - [ ] Windows
 - [X] Mac OSX
 - [X] Linux 

# Description of the bug

The left navigation's text color is white on white background - please note that my OS is in dark mode. Screenshot below
<img width="372" alt="Capture d’écran 2021-05-02 à 11 24 26" src="https://user-images.githubusercontent.com/14168491/116808601-011c0b00-ab3a-11eb-86fc-86da421daaee.png">

# Reproduction

Is the bug reproducible? 
 - [X] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Open the app
2. Navigate through the elements of the nav panel 

…

Expected result:
I can see more than just logos and text of the selected element

Actual result:
I see only logos and not the text of the unselected elements

# Additional information

If you have a backtrace for a crash or a warning, paste it here.

**Comment by 2BeeOrNot** [comment]
You can try changing the navigation panel background color and see if you like the results.
Instructions here: [https://github.com/moneymanagerex/themes](url)

![DarkMode](https://user-images.githubusercontent.com/63317977/116818873-8c49d080-ab3b-11eb-8a81-18ece13dcca4.jpg)

**Comment by fasson** [comment]
Hi there,

Thank you for your reply !

That leads me to another issue:

When I go to menu > tools > theme manager, an issue is raised directly
[image: Capture d’écran 2021-05-02 à 18.41.58.png]

and I can't open any mmextheme extension file (they are greyed and not
clickable)

[image: Capture d’écran 2021-05-02 à 18.42.06.png]

**Comment by 2BeeOrNot** [comment]
I'm not sure how you got to where you are. You got to the point of opening the theme manager so MMEX must have used a theme when opening. The default theme? Then you tried to use the yellow-blue theme when you got the error in the first image? MMEX does some checking of the theme file and if it doesn't like what it finds it will report an error. Before trying to use the yellow-blue theme did you try to edit this theme? I would delete MMEX and reinstall. Then try to use each of the system themes. If all goes well then try and edit the system themes. But never edit the system themes in the MMEX program resource directory. The second image doesn't look right - what is Archive.mmextheme doing in it?

Copy the system theme you want to use as a starting point out of the MMEX program resources folder (in mmex.app on macOS) to your desktop and then rename it to my-yellow-blue.mmextheme for example. Then import it in theme manager and set it as the current theme. User themes are located in the user data folder (where the file mmexini.db3 is located - in macOS it's located in ~Library/Application Support/MoneyManagerEx) in a sub-directory called "themes". Now you can edit the user theme. If you make a mess of it MMEX will complain and use the default theme.

Remember theme files are Zip files with a mmextheme extension instead of Zip extension. If you extract all the files in the theme file you will have to zip them up when you're done editing. If you have a zip file utility you can just extract the file(s) you want to edit and put them back when finished editing.

**Comment by MartinX3** [comment]
I tried all of the default themes.
The text in the left panel is barely to see grey white on a white background.
While clicking on my accounts the entries become white.
If I deselect the MMEX window, the white/hidden entries become grey again.
![grafik](https://user-images.githubusercontent.com/4518092/116859557-7520e680-ac00-11eb-89d3-8da50aa62827.png)

```Money Manager Ex
Version: 1.5.1 64-bit
• Aufbauend auf Apr 25 2021 10:48:35
• Datenbankversion: 7 • (aes128cbc)
MMEX verwendet folgene Unterstützungsprodukte:
• wxWidgets 3.0.5 (wxGTK 3.24)
• wxSQLite3 4.6.0 (SQLite 3.31.1) • RapidJSON 1.1.0
• Lua 5.4.3
• lunasvg 
• libcurl/7.76.1 OpenSSL/1.1.1k zlib/1.2.11 zstd/1.4.9 libidn2/2.3.0 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.9.0 nghttp2/1.43.0
• gettext 0.21
• apexcharts.js
Genutztes Build:
• CMake 3.20.1
• GNU Make 4.3
• GCC 10.2.0 • EndeavourOS rolling 'rolling'
Läuft auf:
• EndeavourOS Linux "rolling"
• Linux 5.11.16-zen1-1-zen x86_64
• German (UTF-8)
• 3200x1980 24bit 95x95ppi ```

**Comment by fasson** [comment]
Ok !! So I think I've isolated the real problem @2BeeOrNot 

My feedback to every of your points for reference, most important feedback at the very end: 

> I'm not sure how you got to where you are. You got to the point of opening the theme manager so MMEX must have used a theme when opening. The default theme? 
Answer : Yes

> Then you tried to use the yellow-blue theme when you got the error in the first image? 
Answer : I tried to upload a new made .mmextheme (small modification of the yellow-blue theme as you advised) and I got that result - which persists now even when I delete - reinstall the dmg package

> MMEX does some checking of the theme file and if it doesn't like what it finds it will report an error. Before trying to use the yellow-blue theme did you try to edit this theme? 
Answer : See above

> I would delete MMEX and reinstall. 
Answer : it does not change anything

> Then try to use each of the system themes. 
Answer : all of them work, but when the theme manager opens, I still get the error message "invalid zip file" with no additional info

> If all goes well then try and edit the system themes. But never edit the system themes in the MMEX program resource directory. The second image doesn't look right - what is Archive.mmextheme doing in it?
Answer : Archive.mmextheme is in a folder "yellow-blue" that I copied from the package to the desktop before modifying it. Archive.mmextheme has been created by compressing all the files.

> Copy the system theme you want to use as a starting point out of the MMEX program resources folder (in mmex.app on macOS) to your desktop and then rename it to my-yellow-blue.mmextheme for example. Then import it in theme manager and set it as the current theme. 
Answer : this resolved the "invalid zip file" issue

> User themes are located in the user data folder (where the file mmexini.db3 is located - in macOS it's located in ~Library/Application Support/MoneyManagerEx) in a sub-directory called "themes". Now you can edit the user theme. If you make a mess of it MMEX will complain and use the default theme.
> 
> Remember theme files are Zip files with a mmextheme extension instead of Zip extension. If you extract all the files in the theme file you will have to zip them up when you're done editing. If you have a zip file utility you can just extract the file(s) you want to edit and put them back when finished editing.

- I copy a theme from the package to desktop => ok
- I rename the extension of this theme on desktop from mmextheme to zip and from zip to mmextheme again => ok
- I unzip the package and compress it again, rename the extension mmextheme => NOT OK - the file is not selectable anymore in the app

**Comment by 2BeeOrNot** [comment]
@fasson 
I'll assume you are using Archive Utility.app to extract the theme files and zip them up again. I don't use it but using it I see where the Archive.mmextheme file name came from.

This worked for me-
I copied my modified version of the default theme, "my-default.mmextheme", to the desktop.
I changed the extension to .zip.
I extracted the files using Archive Utility.app and it created a folder named "my-default".
Now I can edit the files I want to change.
I selected all the files and folders and compressed them.

Archive Utility.app compressed them to a file named Archive.zip.
This is the new my-default theme. But I need to rename it from "Archive.zip" to "my-default.mmextheme"
After renaming I can use the theme manager to import the theme by navigating to the my-default folder on the desktop. MMEX will warn you if the theme already exists and if you want to replace it.

Archive Utility.app is very basic with no UI. I find it useless. I use BetterZip. It's not free but it's great. There are other file managers available, some free, if you search for them.

![MMEX Extract and Compress](https://user-images.githubusercontent.com/63317977/116915244-c5f10900-ac19-11eb-9474-c74a432be0ac.jpg)
![Theme2](https://user-images.githubusercontent.com/63317977/116915701-61827980-ac1a-11eb-84d1-c50c5566aa61.jpg)

**Comment by fasson** [comment]
Thank you @2BeeOrNot - I found the cause. 

Simply, when renaming from .zip to .mmextheme, my Mac still kept the zip extension - the full file was named "my-file.mmextheme.zip" but the last .zip was hidden from the UI... I thought I changed the extension, but I only changed the file name...

Anyway your resolution plan works well when you actually rename your file back to file.mmextheme :-)

I log that trick here for reference as future readers - even _experienced_ Mac users - can meet the same dead end as me !!

I did resolve my issue, thank you for your support !

**Comment by fasson** [comment]
@MartinX3 I did update the description and added Linux - but I assume now it is a non-issue as Themes can be customised following @2BeeOrNot 's detailed steps

**Comment by MartinX3** [comment]
Thank you

I think it is an issue
The design should just work by default and not make stuff invisible or barely to see

**Comment by MartinX3** [comment]
Mentioned https://github.com/moneymanagerex/moneymanagerex/issues/3220

**Comment by Loader009** [comment]
@fasson I agree with @MartinX3, it is an issue.
I have the same thing, barely visible fonts, or the navigation points just disappear if I select a specific entry in the navigator.
This should both not be intended.
